### PR TITLE
Add test for configuration cache bug

### PR DESCRIPTION
**Pull request by staktrace**
This is a test for the scenario in https://github.com/square/wire/issues/2837 which has since been fixed already.

### CODE REVIEW

File: wire-gradle-plugin/src/test/projects/configuration-cache-failure/gradle.properties [hunk 1]
```diff
@@ -0,0 +1,4 @@
+org.gradle.caching=false
+org.gradle.unsafe.configuration-cache=true
+org.gradle.configuration-cache=true
+VERSION_NAME=5.0.0-SNAPSHOT
```
Comment (swankjesse) [inline wire-gradle-plugin/src/test/projects/configuration-cache-failure/gradle.properties:4]: omit this one? I don’t wanna have to keep it up-to-date

**Review by adrw — APPROVED** [review]

**Review by swankjesse — APPROVED** [review]

File: wire-gradle-plugin/src/test/projects/configuration-cache-failure/gradle.properties (hunk 1 above)
Comment (staktrace) [reply wire-gradle-plugin/src/test/projects/configuration-cache-failure/gradle.properties:4]: It's required by the runner: https://github.com/square/wire/blob/b5bf223468acc788941fe088ddd098906a56aae2/wire-gradle-plugin/src/test/kotlin/com/squareup/wire/gradle/WirePluginTest.kt#L1492

**Comment by oldergod** [comment]
Thank you!

File: wire-gradle-plugin/src/test/projects/configuration-cache-failure/gradle.properties (hunk 1 above)
Comment (oldergod) [reply wire-gradle-plugin/src/test/projects/configuration-cache-failure/gradle.properties:4]: We have it somewhere else as well I think, our releasing steps update it so should be good

**Pull request merged**